File: typescript/session3/td/ConsIter.ts
import { Mot } from "./Mot"

export class ConsIter extends Mot {
    var_lettre : string;
    var_reste : Mot;
    var_taille : number;

    public constructor(lettre : string, reste : Mot) {
        super();
        this.var_lettre = lettre;
        this.var_reste = reste;
    }

    public casCons() : boolean {
        return true;
    }

    public lettre() : string {
        return this.var_lettre;
    }

    public reste() : Mot {
        return this.var_reste;
    }

    public taille() : number {
        while(this.casVide()) {
            this.var_taille++;
        }
        return this.var_taille;
    }
}

File: java/cours/src/session3/td/src/ConsIter.java
public class ConsIter implements Mot{
    private char lettre;
    private Mot reste;
    private int taille;

    ConsIter(char lettre, Mot reste) {
        this.lettre = lettre;
        this.reste = reste;
        this.taille =  reste.taille() + 1;
    }

    @Override
    public char lettre() {
        return lettre;
    }

    @Override
    public Mot reste() {
        return reste;
    }

    @Override
    public int taille() {
        return taille;
    }

    @Override
    public boolean casCons() {
        return true;
    }

    public String toString() {
        return "" + lettre + reste.toString();
    }
}
